modules/stochastic_tools/doc/content/source/reporters/ConditionalSampleReporter.md

```diff
@@ -0,0 +1,19 @@
+# ConditionalSampleReporter
+
+!syntax description /Reporters/ConditionalSampleReporter
+
+## Description
+
+This object is a type of [StochasticReporter.md], so it can be used to transfer reporter values from sub-apps into it using [SamplerReporterTransfer.md]. This object is mainly meant for demonstration for eventual active learning algorithms, but could prove useful. Basically, it enables a inputted function to determine if a multiapp solve is "possible" based on sampler values. The parsed function inputted with the parameter [!param](/Reporters/ConditionalSampleReporter/function) should evaluate to 0 or `false` if multiapp run is not possible/necessary. The variables within the function should be associated with a sampler column, which is defined by the pair of parameters [!param](/Reporters/ConditionalSampleReporter/sampler_vars) and [!param](/Reporters/ConditionalSampleReporter/sampler_var_indices). If the inputeed function evaluates to 0 or `false`, the quantity being transferred to this object using [SamplerReporterTransfer.md] with be replaced with [!param](/Reporters/ConditionalSampleReporter/default_value).
```

Comment:
Two reasons. The first being the reason you mentioned. The second being the ability to determine which samples were skipped. If the element was deleted, it would be impossible to determine which sample row exactly caused a skip. Having a default value allows users to filter it when postprocessing.

Reply:
Two reasons. The first being the reason you mentioned. The second being the ability to determine which samples were skipped. If the element was deleted, it would be impossible to determine which sample row exactly caused a skip. Having a default value allows users to filter it when postprocessing.